lib/maropost/contact.rb

```diff
@@ -0,0 +1,91 @@
+require 'active_model'
+
+module Maropost
+  class Contact
+    include ActiveModel::Model
+
+    attr_accessor :id, :email,
+      :travel_weekly, :enews, :travel_especials, :ins_enews, :deals_discounts, :new_member_series, :fleet_contact,
+      :personal_vehicle_reminder, :business_vehicle_reminder, :associate_vehicle_reminder,
+      :vr_reminder_email, :vr_reminder_sms, :vr_reminder_autocall
+
+    def self.find(email)
+      response = Maropost::Api.new(data: { email: email }).find
+      contact = JSON.parse response
+      build contact
+    rescue RestClient::ResourceNotFound => e
+      new(email: email).tap do |contact|
+        contact.errors.add :base, "Email not found"
+      end
+    end
+
+    def create_or_update
```

Comment:
It uses the `contact_data`, which uses the instance attribute values for the contact you call `.create_or_update` on (it can also update the email if you change the value).

The `update_email` could be used without it, but I wanted to make it explicit because it's used differently than the general `update`. We call this when a user changes their email in Gatekeeper (and wherever else). The `create_or_update` is just for subscriptions.